examples/src/main/java/TokenMetadataExample.java

```diff
@@ -0,0 +1,136 @@
+import java.util.Arrays;
+import java.util.Objects;
+
+public class TokenMetadataExample {
+
+    // see `.env.sample` in the repository root for how to specify these values
+    // or set environment variables with the same names
+    private static final AccountId OPERATOR_ID = AccountId.fromString(Objects.requireNonNull(Dotenv.load().get("OPERATOR_ID")));
+    private static final PrivateKey OPERATOR_KEY = PrivateKey.fromString(Objects.requireNonNull(Dotenv.load().get("OPERATOR_KEY")));
+    // HEDERA_NETWORK defaults to testnet if not specified in dotenv
+    private static final String HEDERA_NETWORK = Dotenv.load().get("HEDERA_NETWORK", "testnet");
+
+    private static final PrivateKey METADATA_KEY = PrivateKey.generateED25519();
+
+    private static final byte[] INITIAL_TOKEN_METADATA = new byte[]{1, 1, 1, 1, 1};
+
+    private static final byte[] UPDATED_TOKEN_METADATA = new byte[]{2, 2, 2, 2, 2};
+
+    private Client client;
+
+    public static void main(String[] args) throws Exception {
+        TokenMetadataExample example = new TokenMetadataExample();
+
+        example.updateMutableTokenMetadata();
+
+        example.updateImmutableTokenMetadata();
+
+        example.cleanUp();
+    }
+
+    private TokenMetadataExample() throws Exception {
+        client = ClientHelper.forName(HEDERA_NETWORK);
+
+        // Defaults the operator account ID and key such that all generated transactions will be paid for
+        // by this account and be signed by this key
+        client.setOperator(OPERATOR_ID, OPERATOR_KEY);
+    }
+
+    private void updateMutableTokenMetadata() throws Exception {
+        // create a mutable fungible token with a metadata, but without a metadata key
+        var tokenId = Objects.requireNonNull(
+            new TokenCreateTransaction()
+                .setTokenName("ffff")
+                .setTokenSymbol("F")
+                .setTokenMetadata(INITIAL_TOKEN_METADATA)
+                .setTokenType(TokenType.FUNGIBLE_COMMON) // The same flow can be executed with a TokenType.NON_FUNGIBLE_UNIQUE (i.e. HIP-765)
+                .setTreasuryAccountId(OPERATOR_ID)
+                .setDecimals(3)
+                .setInitialSupply(1000000)
+                .setAdminKey(OPERATOR_KEY)
+                .execute(client)
+                .getReceipt(client)
+                .tokenId
+        );
+
+        System.out.println("Created a mutable token: " + tokenId);
+
+        var tokenInfoAfterCreation = new TokenInfoQuery()
+            .setTokenId(tokenId)
+            .execute(client);
+
+        // check that metadata was set correctly
+        System.out.println("Mutable token's metadata after creation: " + Arrays.toString(tokenInfoAfterCreation.metadata));
+
+        // update token's metadata
+        new TokenUpdateTransaction()
+            .setTokenId(tokenId)
+            .setTokenMetadata(UPDATED_TOKEN_METADATA)
+            .execute(client)
+            .getReceipt(client);
```

Comment:
we should demonstrate the signing of the admin key like in the JS example here: https://github.com/hashgraph/hedera-sdk-js/pull/2210/files#diff-52b14552033ca5d9915427a376d4af66af41bb19a9f65e86ff20502c940cbd63R84

Reply:
Addressed 🙌🏻

Reply:
@SimiHunjan done 🙂